File: Day 08 - Dictionaries and maps/solution.js
function processData(input) {
  //Enter your code here
  const phoneBook = new Map();
  const inputData = input.split("\n");
  const guideSize = Number(inputData.shift()); //3
  // Agregamos cada key:valor en el diccionario
  const queryData = inputData.reduce((target, item, index) => {
    index < guideSize
      ? phoneBook.set(item.split(" ")[0], item.split(" ")[1])
      : target.push(item);

    return target;
  }, []);
  //busca en el diccionario(mapa)
  queryData.forEach((val, index) => {
    console.log(
      phoneBook.get(val) ? `${val}=${phoneBook.get(val)}` : "Not found"
    );
  });
}

File: Day 08 - Dictionaries and maps/solution.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
n = int(input())
phone_guide = {}

for _ in range(n):
    name, number = input().split()
    phone_guide[name] = number

while True: 
    try:
        search_string = str(input())
        if(search_string in phone_guide):
            print(search_string + "=" + phone_guide[search_string])
        else:
            print("Not found")
    except:
        break
